url/Cargo.toml

```diff
@@ -28,18 +28,25 @@ debugger_test = "0.1"
 debugger_test_parser = "0.1"
 
 [dependencies]
-form_urlencoded = { version = "1.2.0", path = "../form_urlencoded" }
-idna = { version = "0.4.0", path = "../idna" }
-percent-encoding = { version = "2.3.0", path = "../percent_encoding" }
+form_urlencoded = { version = "1.2.0", path = "../form_urlencoded", default-features = false, features = ["alloc"] }
+idna = { version = "0.4.0", path = "../idna", default-features = false, features = ["alloc"] }
+percent-encoding = { version = "2.3.0", path = "../percent_encoding", default-features = false, features = ["alloc"] }
 serde = {version = "1.0", optional = true, features = ["derive"]}
+no-std-net = { version = "0.6.0", default-features = false, optional = true }
 
 [features]
-default = []
+default = ["std"]
+std = ["idna/std", "percent-encoding/std", "form_urlencoded/std", "alloc"]
+alloc = []
 # UNSTABLE FEATURES (requires Rust nightly)
 # Enable to use the #[debugger_visualizer] attribute.
 debugger_visualizer = []
 # Expose internal offsets of the URL.
 expose_internals = []
+# For no_std: Allow the use of no_std_net instead of nightly
+no_std_net = ["no-std-net"]
+# For no_std: Use errors_in_core and net_in_core
+unstable = ["idna/unstable"]
```

Comment:
I'm not a big fan of using unstable features in the URL crate 👎 Is there anything we can do to avoid that?

Reply:
The unstable features are optional. It remains totally stable if you either
- use the `no-std-net` feature
- or simply build the library with `std`.
Going forward I hope the libraries get stabilised and the feature flag will lose its purpose.
Personally, I prefer using unstable over some random library for this use case, but all options are there.
I agree it's not ideal, but right now no_std isn't supported at all, so this is a big step up.
Ultimately it's up to you, of course.

Reply:
Adding: Depending on `idna`'s `unstable` feature basically means that you cannot remove that feature without a breaking change in `idna` (because Cargo's feature resolution would require that it is available, at least last time I checked)